pkg/kv/kvserver/scheduler.go

```diff
@@ -297,7 +301,7 @@ func (s *raftScheduler) worker(ctx context.Context) {
 				state.flags |= stateRaftReady
 			}
 		}
-		if state.flags&stateRaftTick != 0 {
+		for t := state.ticks; t > 0; t-- {
```

Comment:
We should have an assertion somewhere that `raftStateTick` also has `ticks > 0`, or guard against it.

Reply:
I wonder if we should just remove `raftStateTick` because it's redundant. You can see in this PR that it's effectively ignored in all the computation, we could just leave `ticks` instead of it.

Reply:
It is redundant, but I think I'd still want to leave it in -- the state flags should cover all states, I think.

Reply:
Thanks! Let's also add an assertion in `worker()`. We can gate it on `RaceEnabled` to avoid any unintended production crashes and overhead (not ideal, but it's a convention we sometimes use).

```
if util.RaceEnabled && (state.flags&stateRaftTick != 0 && state.ticks == 0) || (state.flags&stateRaftTick == 0 && state.ticks != 0) {
    log.Fatalf("stateRaftTick %v with ticks %v", state.flags&stateRaftTick, state.ticks)
}
```

Reply:
Done, with a bit of nesting for better readability.